docker/pytorch/Dockerfile

```diff
@@ -116,12 +118,19 @@ SHELL ["/bin/bash", "-c"]
 # Create a default virtualenv and add it to the path
 ####################################################
 RUN python${PYTHON_VERSION} -m venv --system-site-packages /mosaicml-venv && \
-    /mosaicml-venv/bin/python -m pip install --upgrade pip setuptools && \
+    /mosaicml-venv/bin/python -m pip install --upgrade --no-cache-dir pip setuptools && \
     # Set the path for non-interactive shells
     sed -i 's/PATH=\"/PATH=\"~\/.local\/bin:\/mosaicml-venv\/bin:\/usr\/local\/nvidia\/bin:\/usr\/local\/cuda\/bin:/g' /etc/environment && \
+    echo "export VIRTUAL_ENV=/mosaicml-venv" >> /etc/environment && \
     echo "export PATH=~/.local/bin:/mosaicml-venv/bin:$PATH" >> /etc/profile && \
+    echo "export VIRTUAL_ENV=/mosaicml-venv" >> /etc/profile && \
     # And for interactive shells
-    echo "export PATH=~/.local/bin:/mosaicml-venv/bin:$PATH" >> /etc/bash.bashrc
+    echo "export PATH=~/.local/bin:/mosaicml-venv/bin:$PATH" >> /etc/bash.bashrc && \
+    echo "export VIRTUAL_ENV=/mosaicml-venv" >> /etc/bash.bashrc && \
+    echo "export PATH=~/.local/bin:/mosaicml-venv/bin:$PATH" >> /etc/zshenv && \
+    echo "export VIRTUAL_ENV=/mosaicml-venv" >> /etc/zshenv
+
+ENV VIRTUAL_ENV=/mosaicml-venv
```

Comment:
[nit] I think if you set this before the previous `RUN` you won't need to set the `export VIRTUAL_ENV` in the shell config files and can just invoke `$VIRTUAL_ENV` in the exports for `$PATH`, cleans things up a bit.

Reply:
I found that the `ENV` doesn't always persist depending on what type of shell is launched, so it's safer to set it via the shell files. But yes, putting it before the run helps clean things up